c/driver_manager/adbc_driver_manager.cc

```diff
@@ -138,6 +320,176 @@ struct ManagedLibrary {
     // release() from the DLL - how to handle this?
   }
 
+  AdbcStatusCode GetDriverInfo(const std::string_view driver_name,
+                               const uint16_t load_options, DriverInfo& info,
+                               struct AdbcError* error) {
+    if (driver_name.empty()) {
+      SetError(error, "Driver name is empty");
+      return ADBC_STATUS_INVALID_STATE;
+    }
+
+    std::filesystem::path driver_path(driver_name);
+    const bool allow_relative_paths =
+        load_options & static_cast<uint16_t>(kAdbcAllowRelativePaths);
+    if (driver_path.has_extension()) {
+      if (driver_path.is_relative() && !allow_relative_paths) {
+        SetError(error, "Driver path is relative and relative paths are not allowed");
+        return ADBC_STATUS_INVALID_STATE;
+      }
+
+      // if the extension is .toml, attempt to load the manifest
+      // erroring if we fail
+#ifdef _WIN32
+      if (strncasecmp(driver_path.extension().string().c_str(), ".toml", 5) == 0) {
```

Comment:
Does this match with `.tomlX` too?

Reply:
added a helper function to manage extension comparison to prevent that